roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and inventory_hostname != '127.0.0.1'
```

Comment:
Thanks! Checking whether `ansible_ssh_host` is defined should be enough—it isn't when playbook are running locally.